src/hotspot/cpu/arm/templateInterpreterGenerator_arm.cpp

```diff
@@ -375,79 +373,25 @@ void TemplateInterpreterGenerator::generate_counter_incr(Label* overflow,
-      __ cbz(R1_tmp, no_mdo);
-      // Increment counter in the MDO
-      const Address mdo_invocation_counter(R1_tmp,
-                    in_bytes(MethodData::invocation_counter_offset()) +
-                    in_bytes(InvocationCounter::counter_offset()));
-      const Address mask(R1_tmp, in_bytes(MethodData::invoke_mask_offset()));
-      __ increment_mask_and_jump(mdo_invocation_counter, increment, mask, R0_tmp, Rtemp, eq, overflow);
-      __ b(done);
-    }
-    __ bind(no_mdo);
-    __ get_method_counters(Rmethod, Rcounters, done);
-    const Address mask(Rcounters, in_bytes(MethodCounters::invoke_mask_offset()));
-    __ increment_mask_and_jump(invocation_counter, increment, mask, R0_tmp, R1_tmp, eq, overflow);
-    __ bind(done);
-  } else { // not TieredCompilation
-    const Address backedge_counter(Rcounters,
-                  MethodCounters::backedge_counter_offset() +
-                  InvocationCounter::counter_offset());
-
-    const Register Ricnt = R0_tmp;  // invocation counter
-    const Register Rbcnt = R1_tmp;  // backedge counter
-
-    __ get_method_counters(Rmethod, Rcounters, done);
-
-    if (ProfileInterpreter) {
-      const Register Riic = R1_tmp;
-      __ ldr_s32(Riic, Address(Rcounters, MethodCounters::interpreter_invocation_counter_offset()));
-      __ add(Riic, Riic, 1);
-      __ str_32(Riic, Address(Rcounters, MethodCounters::interpreter_invocation_counter_offset()));
-    }
-
-    // Update standard invocation counters
-
-    __ ldr_u32(Ricnt, invocation_counter);
-    __ ldr_u32(Rbcnt, backedge_counter);
-
-    __ add(Ricnt, Ricnt, InvocationCounter::count_increment);
-
-    __ bic(Rbcnt, Rbcnt, ~InvocationCounter::count_mask_value); // mask out the status bits
-
-    __ str_32(Ricnt, invocation_counter);            // save invocation count
-    __ add(Ricnt, Ricnt, Rbcnt);                     // add both counters
-
-    // profile_method is non-null only for interpreted method so
-    // profile_method != NULL == !native_call
-    // BytecodeInterpreter only calls for native so code is elided.
-
-    if (ProfileInterpreter && profile_method != NULL) {
-      assert(profile_method_continue != NULL, "should be non-null");
-
-      // Test to see if we should create a method data oop
-      // Reuse R1_tmp as we don't need backedge counters anymore.
-      Address profile_limit(Rcounters, in_bytes(MethodCounters::interpreter_profile_limit_offset()));
-      __ ldr_s32(R1_tmp, profile_limit);
-      __ cmp_32(Ricnt, R1_tmp);
-      __ b(*profile_method_continue, lt);
-
-      // if no method data exists, go to profile_method
-      __ test_method_data_pointer(R1_tmp, *profile_method);
-    }
-
-    Address invoke_limit(Rcounters, in_bytes(MethodCounters::interpreter_invocation_limit_offset()));
-    __ ldr_s32(R1_tmp, invoke_limit);
-    __ cmp_32(Ricnt, R1_tmp);
-    __ b(*overflow, hs);
-    __ bind(done);
+  int increment = InvocationCounter::count_increment;
```

Comment:
The small code below seems not to be equivalent replacement for the above code. I would appreciate some explanation on why we change the things. Thanks.

Reply:
It's a just the code that was guarded before by ```if (TieredCompilation)```. It was just the else part that's gotten removed. Since the new policy is base off of tiered we don't need the old style of callbacks. Is there something else that you've noticed?